### PR TITLE
[WIP] Site design refresh

### PR DESCRIPTION
**Pull request by jasonlong**
🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧
### Do not merge

This is a WIP branch with updated tooling and a refreshed site design.

🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧🚧

### Project board

**https://github.com/git/git-scm.com/projects/2**

### Heroku review app URL

**https://git-scm-pr-1179.herokuapp.com**

----

#### Structural

* Responsive layout
* Pro Git book added as top-level section

#### Front end

* Tachyons CSS w/ minimal customizations/additions
* System font stack
* SVG icons / logos
* Updated title tags w/ most specific info at the beginning (for easier scanning in tabs)

#### Performance

* Page weight (TODO)
* CSS stats (TODO)
* JS reduction (TODO)

#### A11y

* Improved color contrast
* `aria-label`s on buttons and permalink anchors.

#### Tooling

* TODO

#### Back end

* TODO

### CODE REVIEW

**Comment by jasonlong** [comment]
:tada: Huge thanks to @appleton for helping get the final Heroku review app kinks worked out! This branch can now be viewed at https://git-scm-pr-1179.herokuapp.com.

**Comment by pedrorijo91** [comment]
@jasonlong anything we can help to keep the momentum? :)

**Comment by jasonlong** [comment]
Sorry for the lack of updates. I hit a couple dead ends on the Manual landing page and then lost some momentum after a trip with my family. Will get going again here ASAP.

**Comment by pedrorijo91** [comment]
sure, no problem! let us know if you need any help on some task, or our thoughts on any stuff ! :)

**Comment by pedrorijo91** [comment]
@jasonlong what's the status on this? :)

**Comment by jasonlong** [comment]
Gah! Yeah sorry for the silence on this. I've been busy with work and recent travels, so have lost momentum. I got bogged down on the index view for the Manual section, so I recently began working on the Book section to regain momentum and hope to have some pushes coming :soon:. I am totally committed to seeing this through - don't worry!

**Comment by peff** [comment]
I wanted to peek at the review deployment for this again, but for some reason it wasn't running. I was able to click-through "create review app" on heroku, but I thought that was supposed to happen automatically.

Anyway, it's running again now, but I wonder if there's something in our heroku config that needs tweaked.

**Comment by pedrorijo91** [comment]
@peff the deployment is killed after some time without no updates (not sure if configurable). I think the main idea is to avoid increasing costs on stale PRs

**Comment by peff** [comment]
Ah, OK, I wondered if it might be something like that. All is behaving as expected, then. :sweat_smile:

**Comment by jasonlong** [comment]
Hey everyone, I'm so sorry I haven't made any progress on this in ages. Things at work have been hectic and I just have found the time I've wanted to devote to this redesign.

The projects I'm working on will ship at GitHub Universe in October and I anticipate having some more breathing room after that. I'd like to make a final push at that time to get this work completed. Thanks for bearing with me.

File: app/views/doc/_versions.html.erb [hunk 1]
```diff
@@ -1,36 +1,43 @@
...
+              <span class="b dib w3"><%= v[:name] %></span>
+              <span class="dn dib-l w3 pr2">
+                <% v[:diff][0].times do %>
+                  <img src="/images/icons/green-dot.png" class="nl-1px" alt="Green dot" />
+                <% end %>
+                <% v[:diff][1].times do %>
+                  <img src="/images/icons/red-dot.png" class="nl-1px" alt="Green dot" />
+                <% end %>
+                <% v[:diff][2].times do %>
+                  <img src="/images/icons/grey-dot.png" class="nl-1px" alt="Green dot" />
```
[the remark this answers is not in the document]
Comment (jasonlong) [reply app/views/doc/_versions.html.erb]: Gah, thanks!

File: Gemfile.lock [hunk 2]
```diff
@@ -138,6 +119,10 @@ GEM
       pry (~> 0.10)
     public_suffix (3.0.2)
     puma (3.11.3)
+<<<<<<< HEAD
```
Comment (connorshea) [inline Gemfile.lock]: Conflict markers are still here :)

**Comment by pedrorijo91** [comment]
@jasonlong free to pick this up again? how can we help? :)

**Comment by jasonlong** [comment]
@pedrorijo91 Sorry again for the lack of progress here. I keep shooting myself in the foot here since each time I finally find some time to spend on this project my dev setup is broken and by the time I've gotten things fixed up, I get really busy with work again.

I'm not sure how easy it would be, but maybe someone could help get this branch updated with all of the updates to `master` over the past few months? If there are conflicts that need my context, those can be left for me to check into.

**Comment by pedrorijo91** [comment]
totally understand @jasonlong :sweat_smile: 

If I remember correctly, there weren't any major changes recently. A few new books/GUIs, fixing the ES client, and some work on book translations.

will have a look during this week to help you :)

**Comment by pedrorijo91** [comment]
@jasonlong sorry for taking to much time, last week was chaotic. Started merging master into v2. Still halfway, but will look at it in a couple of hours to finish it still today

**Comment by prasadgujar** [comment]
#1188 this is resolved under this PR @jasonlong @pedrorijo91 @peff @rybak @connorshea ?

**Comment by peff** [comment]
@jasonlong There as discussion a few weeks at the last git contributors summit about setting up a community blog, that could be accessible at `blog.git-scm.com` or similar. Do you have any recommendations on styling to make it look consistent with the rest of the site? I.e., should people pull style elements from the existing site, or from this PR? Do you have an idea of whether you might pick this PR up again?

**Comment by jasonlong** [comment]
Hey @peff, I should be able to help with the blog styling. Is the plan to use a standalone site? I could extract the basic styling from the "new" design into something appropriate for the blog.

And yes, I still want to see this redesign through. I'm struggling to carve out time to be able to deeply get into it.

**Comment by peff** [comment]
@jasonlong Sorry for the slow reply. Yeah, the plan is to keep it standalone, to keep it away from the Rails complexity in the current site (and also it would probably have a larger set of people contributing to it). Right now there's a bare-bones Hugo setup at https://gitlab.com/git-scm/blog, but there are no posts yet.

**Comment by jamesramsay** [comment]
@jasonlong I'm working on trying to get a minimal version of the blog up and running linked by @peff. If you have a mockup of the new site you are working on, I could take that as inspiration so that there is a some similarity.

**Comment by pedrorijo91** [comment]
there used to be a live app with the new UI, but it seems there's an error deploying to heroku currently..

**Comment by olegKusov** [comment]
I have some ideas about git-scm. 
1. New minimalistic and responsive design. I cant help with it.
2. New ideas with documentation. Git is not easy. And it will be cool if we add some interactive descriptions with live branch creation tool on web page. For example , user reads something about "git revert" and on the page he can create branches and play with them with some little lessons from git-scm. I think it will be cool. Know git-scm is just hard readable documentation. And that's why we have a lot of blogs about git. We need to make official documentation easy for noobs.

**Comment by felipec** [comment]
Is there any link to see this redesign in action?

**Comment by pedrorijo91** [comment]
closing as it's stale for a long time

**Comment by vitaly-zdanevich** [comment]
So, no new design with dark theme? :(

**Pull request closed without merging**